sycl/test/xocc_tests/simple_tests/internal_defines.cpp

```diff
@@ -0,0 +1,63 @@
+// RUN: true
+
+/*
+
+  This test is checking if the implementation defined defines created by the
+  compiler are being defined correctly for the device and host compilation.
+
+  Some components like the SPIR intrinsics and xilinx includes rely on these
+  flags.
+
+  When compiling for a Xilinx device the following defines should be defined:
+
+  __SYCL_XILINX_ONLY__ : defined on both the host and device
+  __SYCL_SPIR_ONLY__ : defined only on the device
+
+  When compiled for anything but a Xilinx device at the moment both flags should
+  be undefined.
+*/
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+#include "../utilities/device_selectors.hpp"
+
+using namespace cl::sycl;
+
+int main() {
+  selector_defines::CompiledForDeviceSelector selector;
+  queue q {selector};
+
+#ifdef __SYCL_SPIR_DEVICE__
+        printf("__SYCL_SPIR_DEVICE__ defined on host \n");
```

Comment:
Why `printf`? First this is a C++ program and even in C you should use `puts` because there is no format...

Reply:
Habitual usage from OpenCL kernels